doc/sphinx/language/gallina-extensions.rst

```diff
@@ -966,12 +954,35 @@ are now available through the dot notation.
 .. exn:: Signature components for label @ident do not match.
     :undocumented:
 
-.. exn:: This is not the last opened module.
-    :undocumented:
-
-.. exn:: This is not the last opened module type.
+.. exn:: The field @ident is missing in @qualid.
    :undocumented:
 
+.. |br| raw:: html
+
+    <br>
+
+.. note::
+
+  #. Interactive modules and module types can be nested.
+  #. Interactive modules and module types can't be defined inside of :ref:`sections<section-mechanism>`.
+     Sections can be defined inside of interactive modules and module types.
+  #. Hints and notations (:cmd:`Hint` and :cmd:`Notation` commands) can also appear inside interactive
+     modules and module types. Note that with module definitions like:
+
+     :n:`Module @ident__1 : @module_type := @ident__2.`
+
+     or
+
+     :n:`Module @ident__1 : @module_type.` |br|
+     :n:`Include @ident__2.` |br|
+     :n:`End @ident__1.`
+
+     hints and the like valid for :n:`@ident__1` are the ones defined in :n:`@module_type`
+     rather then those defined in :n:`@ident__2` (or the module body).
+
+  #. Within an interactive module type definition, the :cmd:`Parameter` command declares a
+     constant instead of definining a new axiom (which it does when not in a module type definition).
```

Comment:
Looks good but the documentation of the `Inline` clause should also be put here.

Reply:
Unclear what you want me to add (or move?) here.

Reply:
Cf. the comment just below "This is what I'm talking about".